### PR TITLE
Fix QuerySchedulerFactory to plug in custom scheduler

### PR DESCRIPTION
**Pull request by Jackie-Jiang**
Fix QuerySchedulerFactory so that custom scheduler can be plugged in

### CODE REVIEW

**Review by richardstartin — APPROVED** [review]

File: pinot-core/src/main/java/org/apache/pinot/core/query/scheduler/QuerySchedulerFactory.java [hunk 1]
```diff
@@ -89,12 +96,16 @@ public static QueryScheduler create(PinotConfiguration schedulerConfig, QueryExe
...
-      Constructor<?> constructor =
-          Class.forName(className).getDeclaredConstructor(Configuration.class, QueryExecutor.class);
+      Constructor<?> constructor = Class.forName(className)
+          .getDeclaredConstructor(PinotConfiguration.class, QueryExecutor.class, ServerMetrics.class,
+              LongAccumulator.class);
       constructor.setAccessible(true);
-      return (QueryScheduler) constructor.newInstance(schedulerConfig, queryExecutor);
+      QueryScheduler queryScheduler =
+          (QueryScheduler) constructor.newInstance(schedulerConfig, queryExecutor, serverMetrics, latestQueryTime);
+      LOGGER.info("Using {} scheduler", queryScheduler.name());
```
Comment (mcvsubbu) [inline pinot-core/src/main/java/org/apache/pinot/core/query/scheduler/QuerySchedulerFactory.java:107]: maybe move this log statement to line 81 (and perhaps use a single log statement at the end of the switch there?)

File: pinot-core/src/main/java/org/apache/pinot/core/query/scheduler/QuerySchedulerFactory.java [hunk 2]
```diff
@@ -89,12 +96,16 @@ public static QueryScheduler create(PinotConfiguration schedulerConfig, QueryExe
 
   @Nullable
   private static QueryScheduler getQuerySchedulerByClassName(String className, PinotConfiguration schedulerConfig,
-      QueryExecutor queryExecutor) {
+      QueryExecutor queryExecutor, ServerMetrics serverMetrics, LongAccumulator latestQueryTime) {
     try {
-      Constructor<?> constructor =
-          Class.forName(className).getDeclaredConstructor(Configuration.class, QueryExecutor.class);
+      Constructor<?> constructor = Class.forName(className)
```
Comment (mcvsubbu) [inline pinot-core/src/main/java/org/apache/pinot/core/query/scheduler/QuerySchedulerFactory.java:101]: Should we not be using PluginManager for class loads?

File: pinot-core/src/main/java/org/apache/pinot/core/query/scheduler/QuerySchedulerFactory.java (hunk 1 above)
Comment (Jackie-Jiang) [reply pinot-core/src/main/java/org/apache/pinot/core/query/scheduler/QuerySchedulerFactory.java]: Done

File: pinot-core/src/main/java/org/apache/pinot/core/query/scheduler/QuerySchedulerFactory.java (hunk 2 above)
Comment (Jackie-Jiang) [reply pinot-core/src/main/java/org/apache/pinot/core/query/scheduler/QuerySchedulerFactory.java]: Good point. Done

**Review by mcvsubbu — APPROVED** [review]

**Pull request merged**